### PR TITLE
BLKD Frame Writer and JSON deserialize helper

### PR DESCRIPTION
**Pull request by chenliu0831**
1. Add IR images and depthToCamera to savedItemTypes
2. A BLKD Frame writer
3. JSON Deserialize helper

### CODE REVIEW

File: Smithers.Serialization/SavedItemType.cs [hunk 1]
```diff
@@ -68,6 +68,9 @@ public class SavedItemType
         public static readonly SavedItemType SKELETON = new SavedItemType("Skeleton");
         public static readonly SavedItemType BODY_INDEX = new SavedItemType("BodyIndex");
         public static readonly SavedItemType HD_FACE = new SavedItemType("HDFace");
+        public static readonly SavedItemType DEPTH_TO_CAMERA_MAPPING = new SavedItemType("DepthToCamera");
+        public static readonly SavedItemType IR_RIGHT_IMAGE = new SavedItemType("IR_RIGHT");
+        public static readonly SavedItemType IR_LEFT_IMAGE = new SavedItemType("IR_LEFT");
```
Comment (paulmelnikow) [inline Smithers.Serialization/SavedItemType.cs]: Hmm, these types don't quite make sense for Kinect. Can you move these declarations to the consuming code instead?

File: Smithers.Sessions/JSONHelper.cs [hunk 2]
```diff
@@ -107,6 +107,40 @@ public void Serialize(object data, string path)
...
+                    {
+                        result = new T();
+                    }
+
+                    return result;
+                }
+            }
+            catch (IOException)
+            {
+                return new T();
```
Comment (paulmelnikow) [inline Smithers.Sessions/JSONHelper.cs]: Failing silently seems like a bad idea. I realize this is copied and pasted from below, but would rather we clean it up.

File: Smithers.Serialization/Writers/BitmapWriter.cs [hunk 3]
```diff
@@ -67,6 +68,20 @@ public override void Write(Stream stream)
...
+        public override string FileExtension { get { return ".blkd"; } }
+
+        protected abstract Blkd BlkdSource { get; }
+
+        public override void Write(Stream stream)
+        {
+            Smithers.Serialization.Formats.Blkd.Save(this.BlkdSource, stream);
+        }
+    }
+
```
Comment (paulmelnikow) [inline Smithers.Serialization/Writers/BitmapWriter.cs:84]: LGTM

File: Smithers.Sessions/JSONHelper.cs (hunk 2 above)
Comment (chenliu0831) [inline Smithers.Sessions/JSONHelper.cs]: @paulmelnikow So should we throw the `IOException` and let the caller handle it?

Comment (paulmelnikow) [inline Smithers.Sessions/JSONHelper.cs]: Yeah, letting the exception propagate seems fine.

**Comment by paulmelnikow** [comment]
Hey @chenliu0831 let's get this merged. Can you squash it into a single commit?

**Comment by chenliu0831** [comment]
ahh.. How can I merge it? Seems I don't have write access

**Pull request merged**